### PR TITLE
Updating stack(C#) to be a proper stack

### LINKED ISSUE #517: Issue in stack(C#) implementation
#### This issue is about [Algo/DS Name](link to resource for the Algo/DS).

- [ ] Issue Title is in form `Algo/DS Name [Language]`.
- [x] I searched or browsed the repo’s other issues to ensure this is not a duplicate.
- [ ] This Algo/DS is worth adding to this repository.
- [ ] Assign this issue to me (I want to work on this).
	- [ ] I ensure that I am not already assigned to 2 or more issues.
- [x] I will send my PR only after I'm assigned to this issue by any maintainer.
- [ ] I understand that I will be unassigned from this issue if I fail to send a PR for this within a week.

`Rotate` and `ToString` method traverses the stack without popping out it's elements. This is not allowed in Stack. A similar discussion is done in issue #125 . Please look into it. @singhpratyush @Monal5031 @aashutoshrathi @yashLadha @aviaryan .

### PR DESCRIPTION
**Pull request by PlatanoBailando**
Fixes #517

#### By submitting this pull request I confirm I've read and complied with the below declarations.

- [x] I have read the [Contribution guidelines](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md) and I am confident that my PR reflects them.
- [X] I have followed the [coding guidelines](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md#cs) for this project.
- [X] My code follows the [skeleton code structure](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md#sample).
- [X] This pull request has a descriptive title. For example, `Added {Algorithm/DS name} [{Language}]`, not `Update README.md` or `Added new code`.
- [X] This pull request will be closed if I fail to update it even once in a continuous time span of 7 days.

### CODE REVIEW

**Comment by singhpratyush** [comment]
Hi! Thanks for showing interest in contributing to this repository.

Please make sure that you have ticked the points mentioned in PR description correctly.

Thanks again!

File: stack/Stack.cs [hunk 1]
```diff
@@ -58,92 +49,9 @@ public void Push(T data)
...
-    }
-
-    /**
-    * gets the n top elements and rotates them
-    *
-    * if left == false then it will move the last element to behind the nth element(from the top)
-    *
-    * if left == true it will move the nth element(from the top) to the top
-    */
-    public void Rotate(int n, bool left)
```
Comment (singhpratyush) [inline stack/Stack.cs]: The issue mentions that this method should use stack operations to rotate the numbers. But you have removed the method here, which is not expected. Please use stack methods to accomplish this method.

File: stack/Stack.cs [hunk 2]
```diff
@@ -58,92 +49,9 @@ public void Push(T data)
...
-            for (int i = 0; i < n; i++)
-            {
-                node = node.Previous;
-            }
-
-            stack.AddBefore(node.Next, last);
-        }
-    }
-
-    public override string ToString()
```
Comment (singhpratyush) [inline stack/Stack.cs]: Same here.

File: stack/Stack.cs (hunk 1 above)
Comment (PlatanoBailando) [reply stack/Stack.cs]: Ah ok, I misunderstood the ticket.  I will update my PR.

**Pull request merged**